ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -30,15 +30,35 @@
 
 #include <fstream>
 
-gcp_connection::gcp_connection(const std::string &bucket_name)
-    : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name)
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+
+gcp_connection::gcp_connection(const std::string &bucket_name, const std::string &prefix)
```

Comment:
It looks like `_prefix` is set once and lasts the entire lifetime of the connection. Is this intentional?

I also noticed in earlier commits we had a separate call to `_gcp_client.ListObjects` that didn't provide a `gcs::Prefix` argument when no prefix was provided. This seems like a better choice to me as there's likely optimisations in  `_gcp_client.ListObjects(_bucket_name)` that we lose when passing in the extra argument `_gcp_client.ListObjects(_bucket_name, gcs::Prefix("")`

Reply:
Hmm good point @ajmorton, I think it may provide performance optimisations without the Prefix argument, but I would think that the SDK would have dealt with this as well. Nonetheless I am okay with the optimisations not being optimal for that scenario, because in the real world scenario a prefix will be applied most of the time.